examples/fast_neural_style/neural_style.py

```diff
@@ -0,0 +1,216 @@
+# coding: utf-8
+from __future__ import print_function, division
+
+import argparse
+import os
+import sys
+
+import numpy as np
+import random
+import torch
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torchvision import datasets
+from torchvision import transforms
+
+from ignite.engine import Engine, Events
+from ignite.handlers import ModelCheckpoint
+
+import utils
+from transformer_net import TransformerNet
+from vgg import Vgg16
+from handlers import Progbar
+
+from collections import OrderedDict
+
+
+def check_paths(args):
+    try:
+        if args.checkpoint_model_dir is not None and not (os.path.exists(args.checkpoint_model_dir)):
+            os.makedirs(args.checkpoint_model_dir)
+    except OSError as e:
+        raise OSError(e)
+
+
+def check_manual_seed(args):
+    seed = args.seed or random.randint(1, 10000)
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+
+
+def check_dataset(args):
+    transform = transforms.Compose([
+        transforms.Resize(args.image_size),
+        transforms.CenterCrop(args.image_size),
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+
+    if args.dataset in {'folder', 'mscoco'}:
+        train_dataset = datasets.ImageFolder(args.dataroot, transform)
+    elif args.dataset == 'test':
+        train_dataset = datasets.FakeData(size=args.batch_size, image_size=(3, 32, 32),
+                                          num_classes=1, transform=transform)
+    else:
+        raise RuntimeError("Invalid dataset name: {}".format(args.dataset))
+
+    train_loader = DataLoader(train_dataset, batch_size=args.batch_size, shuffle=True, num_workers=0)
+
+    return train_loader
+
+
+def train(args):
+    device = torch.device("cuda" if args.cuda else "cpu")
+
+    train_loader = check_dataset(args)
+    transformer = TransformerNet().to(device)
+    optimizer = Adam(transformer.parameters(), args.lr)
+    mse_loss = torch.nn.MSELoss()
+
+    vgg = Vgg16(requires_grad=False).to(device)
+    style_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+
+    style = utils.load_image(args.style_image, size=args.style_size)
+    style = style_transform(style)
+    style = style.repeat(args.batch_size, 1, 1, 1).to(device)
+
+    features_style = vgg(utils.normalize_batch(style))
+    gram_style = [utils.gram_matrix(y) for y in features_style]
+
+    running_avgs = OrderedDict()
+
+    def step(engine, batch):
+
+        x, _ = batch
+        x = x.to(device)
+
+        n_batch = len(x)
+
+        transformer.zero_grad()
```

Comment:
`transformer.zero_grad()` and `optimizer.zero_grad()` should be the same AFAIK, but can we change this to `optimizer.zero_grad()` to be consistent with the other examples?